src/AbstractDifferentiation.jl

```diff
@@ -1,5 +1,332 @@
 module AbstractDifferentiation
 
-# Write your package code here.
+using LinearAlgebra
 
+export AD
+
+const AD = AbstractDifferentiation
+
+abstract type AbstractBackend end
+abstract type AbstractFiniteDifference <: AbstractBackend end
+abstract type AbstractForwardMode <: AbstractBackend end
+abstract type AbstractReverseMode <: AbstractBackend end
+
+struct HigherOrderBackend{B} <: AbstractBackend
+    backends::B
+end
+reduceorder(b::AbstractBackend) = b
+function reduceorder(b::HigherOrderBackend)
+    return HigherOrderBackend(reverse(Base.tail(reverse(b.backends))))
+end
+lowest(b::AbstractBackend) = b
+lowest(b::HigherOrderBackend) = b.backends[end]
+secondlowest(b::HigherOrderBackend) = lowest(reduceorder(b))
```

Comment:
I don't get this part

Reply:
It's an over-complicated way to get `b.backends[end-1]`. I was trying to be generic but I don't think generic helps here.

Reply:
The lowest-level backend is `b.backends[end]`. The second lowest is `b.backends[end-1]`. In forward-over-reverse, the lowest is reverse and the second lowest is forward.